matlab/schemas/+stimulus/Matisse2.m

```diff
@@ -0,0 +1,159 @@
+            cond.ori = 30;
+            cond.outer_ori_delta = 90;
+            cond.coherence = 2.5;
+            cond.aperture_x = 0.2;
+            cond.aperture_y = 0.1;
+            cond.aperture_r = 0.2;
+            cond.aperture_transition = 0.1;
+            cond.annulus_alpha = 0.0;
+            cond.outer_contrast = 1;
+            cond.inner_contrast = 1;
+            cond.outer_speed = 0.2;
+            cond.inner_speed = 0.2;
+            tic
+            cond = stimulus.Matisse2.make(cond);
+            toc
+            
+            v = VideoWriter('Matisse2', 'MPEG-4');
+            v.FrameRate = cond.fps;
+            v.Quality = 100;
+            open(v)
+            writeVideo(v, permute(cond.movie, [1 2 4 3]));
+            close(v)
+        end
+        
+        
+        function cond = make(cond)
+            assert(~verLessThan('matlab','9.1'), 'Please upgrade MATLAB to R2016b or better')
+            nframes = round(cond.duration*cond.fps);
+            img = randn(round(cond.pattern_width/cond.pattern_aspect), cond.pattern_width);
+            outer = upscale(img, cond.pattern_upscale, cond.ori + cond.outer_ori_delta, ...
+                cond.coherence, nframes, cond.outer_speed*cond.pattern_upscale*cond.pattern_width/cond.fps);
+            inner = upscale(img, cond.pattern_upscale, cond.ori, ...
+                cond.coherence, nframes, cond.inner_speed*cond.pattern_upscale*cond.pattern_width/cond.fps);
+            img = aperture(inner*cond.inner_contrast, outer*cond.outer_contrast, ...
+                cond.aperture_x, cond.aperture_y, cond.aperture_r, cond.aperture_transition, cond.annulus_alpha);
+            cond.movie = uint8(img*256+127.5);
+        end
+    end
+    
+    methods
+        function showTrial(self, cond)
+            % verify that pattern parameters match display settings
+            assert(~isempty(self.fps), 'Cannot obtain the refresh rate')
+            assert(abs(self.fps - cond.fps)/cond.fps < 0.05, 'incorrect monitor frame rate')
+            assert((self.rect(3)/self.rect(4) - cond.pattern_aspect)/cond.pattern_aspect < 0.05, 'incorrect pattern aspect')
+            
+            % blank the screen if there is a blanking period
+            opts.clearScreen = true;
+            opts.checkDroppedFrames = false;
+            if cond.pre_blank_period>0
+                opts.logFlips = false;
+                self.flip(opts)
+                WaitSecs(cond.pre_blank_period);
+            end
+            
+            % play movie
+            opts.logFlips = true;
+            for i=1:size(cond.movie,3)
+                tex = Screen('MakeTexture', self.win, cond.movie(:,:,i));
+                Screen('DrawTexture',self.win, tex, [], self.rect)
+                self.flip(opts)
+                opts.checkDroppedFrames = true;
+                Screen('close',tex)
+            end
+        end
+    end
+end
+
+
+function img = upscale(img, factor, ori, coherence, nframes, speed)
```

Comment:
Did you mean "indented"?  I use MATLAB's indentation rules, which are enforced when you mark all text (Ctrl-A) and then press Ctrl-I.   These are function that are external to the class and they are not indented in MATLAB.